libs/libvtrutil/src/vtr_ndmatrix.h

```diff
@@ -36,7 +36,6 @@ class NdMatrixProxy {
     NdMatrixProxy<T, N>& operator=(const NdMatrixProxy<T, N>& other) = delete;
 
     const NdMatrixProxy<T, N - 1> operator[](size_t index) const {
-        VTR_ASSERT_SAFE_MSG(index >= 0, "Index out of range (below dimension minimum)");
```

Comment:
Why does this assert go away?

Reply:
GCC warns (correctly) that size_t can't be less than zero as it is an unsigned type, so the assertion must always be true (and the assertion is redundant).

(That assertion actually came as legacy from an earlier version of the NdMatrix which allowed indicies to start from non-zero values.)

Reply:
I was sure that size_t was signed but it seems like that is false -- https://stackoverflow.com/questions/1089176/is-size-t-always-unsigned
> Yes, size_t is guaranteed to be an unsigned type. According to the standard it is unsigned, however I recall that some older implementations used a signed type for the typedef.

Depending on how paranoid we want to be -- maybe we just need a compiler assert somewhere that `size_t` is unsigned?

Reply:
> I was sure that size_t was signed but it seems like that is false

Perhaps you were thinking of `ssize_t` from POSIX?

> maybe we just need a compiler assert somewhere that `size_t` is unsigned?

I believe that is guaranteed by the C++ standard so it seems redundant (see: https://en.cppreference.com/w/cpp/types/size_t).

Reply:
Looks like you are right! Thanks.